packages/aws-cdk-lib/core/lib/metadata-resource.ts

```diff
@@ -42,15 +37,20 @@ export function addConstructMetadata(scope: Construct, props: any): void {
      * Errors are ignored here.
      * We do not want the metadata parsing to block users to synth or
      * deploy their CDK application.
-     * 
+     *
      * Without this, it will just fall back to the previous metadata
      * collection strategy.
      */
   }
 }
 
 function addMetadata(scope: Construct, type: MetadataType, props: any): void {
-  scope.node.addMetadata(type, redactTelemetryDataHelper(props));
```

Comment:
So before we always redact something but now if the telemetry is disabled, we do not redact anything?

Reply:
No this code was introduced by my previous PR (and this behaviour is not public yet). If FF is off, we do not even add this metadata now.